server/client.go

```diff
@@ -457,13 +457,15 @@ func (c *client) processConnect(arg []byte) error {
 		}
 
 		// Client-enabled tracing. Check permissions if applicable.
+		c.mu.Lock()
 		if c.opts.Trace {
 			if srv.opts.OptInTrace || (c.perms != nil && c.perms.trace) {
 				c.trace = true
 			} else {
 				c.sendErr("Permissions Violation for trace option")
 			}
 		}
+		c.mu.Unlock()
```

Comment:
You can't do that. `sendErr` grabs the client's lock, so this would deadlock. That means that this part is not tested.